barretenberg/cpp/src/barretenberg/vm/avm_trace/avm_trace.cpp

```diff
@@ -368,29 +370,47 @@ void AvmTraceBuilder::set(uint128_t val, uint32_t dst_offset, AvmMemoryTag in_ta
  * @param src_offset Offset of source memory cell
  * @param dst_offset Offset of destination memory cell
  */
-void AvmTraceBuilder::op_mov(uint32_t const src_offset, uint32_t const dst_offset)
+void AvmTraceBuilder::op_mov(bool indirect, uint32_t const src_offset, uint32_t const dst_offset)
```

Comment:
This would need to be a `uint_8` or two bool flags to control the indirect-edness of either the src or dest addresses.

We can also decompose the indirect byte in the deserialisation and just have bools here or handle the bit-decomposition here.

Reply:
According to the yellow paper, this is a flag which toggles "each argument offset". Indirect memory concept would accommodate a per-argument indirect enablement easily. I would keep the boolean here as long as there is no need to extend. Note that the de-serialization layer is already parsing uint8_t.

This would be trivial to replace the boolean to uint8_t to support more flavours of indirectness.

Reply:
Thanks for this. I thought indirectness is flag which is applied for all offset operands. I will adapt the code then.